pennylane/templates/state_preparations/basis.py

```diff
@@ -52,20 +52,32 @@ def circuit(basis_state):
 
     def __init__(self, basis_state, wires, do_queue=True, id=None):
 
-        shape = qml.math.shape(basis_state)
+        # check if the `basis_state` param is batched
+        batched = len(qml.math.shape(basis_state)) > 1
 
-        if len(shape) != 1:
-            raise ValueError(f"Basis state must be one-dimensional; got shape {shape}.")
+        state_batch = basis_state if batched else [basis_state]
 
-        n_bits = shape[0]
-        if n_bits != len(wires):
-            raise ValueError(f"Basis state must be of length {len(wires)}; got length {n_bits}.")
+        for i, state in enumerate(state_batch):
+            shape = qml.math.shape(state)
 
-        # we can extract a list here, because embedding is not differentiable
-        basis_state = list(qml.math.toarray(basis_state))
+            if len(shape) != 1:
+                raise ValueError(
+                    f"Basis states must be one-dimensional; state {i} has shape {shape}."
+                )
 
-        if not all(bit in [0, 1] for bit in basis_state):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {basis_state}")
+            n_bits = shape[0]
+            if n_bits != len(wires):
+                raise ValueError(
+                    f"Basis states must be of length {len(wires)}; state {i} has length {n_bits}."
+                )
+
+            # we can extract a list here, because embedding is not differentiable
+            state = list(qml.math.toarray(state))
```

Comment:
Note on a subtlety: in the original implementation, the `basis_state` object itself was updated in this step and it was converted into a list before it was passed on to the parent constructor (line 70).

Here, the temporary `state` object is being converted, rather than `basis_state`. Would we want to convert `basis_state` to a list?

Reply:
Hmmm, good question. It doesn't look like it's necessary since the `batch_param` decorator already converts the parameters into nested lists:
```
unbatched_params = [[] for i in range(batch_dim)]

for p in params:
    for i in range(batch_dim):
        try:
            unbatched_params[i].append(p[i])
        except IndexError:
            ...
```
That said, I did try versions of the code with/without converting `state` into a list and with/without converting `basis_state` into a list, and all versions work perfectly fine.